In [30]:
%matplotlib inline
import matplotlib.pyplot as plt
import numpy as np

j=1j

sx=np.array([[0,1],[1,0]])
ex=np.linalg.eig(sx) 

sy=np.array([[0,-j],[j,0]])
ey=np.linalg.eigh(sy) 

sz=np.array([[1,0],[0,-1]])
ez=np.linalg.eig(sz)

ez,ex,ez

((array([ 1., -1.]),
  array([[1., 0.],
         [0., 1.]])),
 (array([ 1., -1.]),
  array([[ 0.70710678, -0.70710678],
         [ 0.70710678,  0.70710678]])),
 (array([ 1., -1.]),
  array([[1., 0.],
         [0., 1.]])))

sx의 아이겐벡터는 $v_1$=$(\begin {matrix} 1 \\\ 0 \end{matrix})$, $v_2$=(0,1)이며 아이겐밸류는 (1,-1)이다.
                     
sy의 아이겐벡터는 $v_1$=$(\begin {matrix} 0.70710678 \\\ 0.70710678 \end{matrix})$ = $(\begin {matrix} 1 \\\ 1 \end{matrix})$, $v_2$=(-0.70710678,0.7071678)=(-1,1)이며 아이겐밸류는 (1,-1)이다.
                     
sz의 아이겐벡터는 $v_1$=$(\begin {matrix} 1 \\\ 0 \end{matrix})$, $v_2$=(0,1)이며 아이겐밸류는 (1,-1)이다.
                      

### $\frac{k}{mw^2}$$(\begin {matrix} 2 & -1 \\\ -1 &   2 \end{matrix})$ $(\begin {matrix} x_1 \\\ y_1 \end{matrix})$ = $\lambda$ $(\begin {matrix} x_1 \\\ y_1 \end{matrix})$
위의 식에서 vector$(\begin {matrix} x_1 \\\ y_1 \end{matrix})$이 같기 때문에  $\lambda$=$\frac{k}{mw^2}$$(\begin {matrix} 2 & -1 \\\ -1 & 2 \end{matrix})$라고 할 수 있다.  
이를 통해 $\lambda$를 구해보면 다음과 같다.

In [31]:
m,k=1,1
w=np.sqrt(k/m)

a=k/(m*(w**2))

ld=a*np.array([[2,-1],[-1,2]])
ed=np.linalg.eig(ld)

ed

(array([3., 1.]),
 array([[ 0.70710678,  0.70710678],
        [-0.70710678,  0.70710678]]))

ed의 아이겐벡터는 $v_1$=$(\begin {matrix} 0.70710678 \\\ -0.70710678 \end{matrix})$ = $(\begin {matrix} 1 \\\ -1 \end{matrix})$, $v_2$=(0.70710678, 0.70710678) = (1,1)이며 아이겐밸류는 (3,1)이다.

구한 아이겐밸류를 각각의 람다 값에 대입하면  
$\lambda$=1일 경우, $w^2$=$\frac{k}{m}$이므로 $w_1$=$\sqrt \frac{k}{m}$가 되며, -$kx_1$=-2$kx_1$+$kx_2$, 
$kx_1$=$kx_2$, $x_1$=$x_2$가 된다.   
$\lambda$=3일 경우, $3w^2$= $\frac {k}{m}$이므로 $w_2$=$\sqrt \frac{3k}{m}$가 되며, -3$kx_2$=k$x_1$-2$kx_2$,  
$kx_2$=-$kx_1$ $x_2$=-$x_1$이 된다.  
   따라서 $(\begin {matrix} 1 & -1 \\\ 1 &   1 \end{matrix})$을 구할 수 있다.

여기서 구한 eigen vector은 operation에 대해서 계산해줄 때 제일 편하게 기준이 되는 축을 잡아주는 것이다. 따라서 eigen vector은 어떠한 변화를 하던지간에 축에 대해서 M이라고 하는 operation에 의한 변환은 모든 변환들이 축에 대해서 상수배만큼 곱해주는 것으로 바뀌어 모든 값들을 나타낼 수 있다.

그리고 원래의 vector에 대해서 몇 배가 되는지를 나타내주는 것이 eigen value이다.